pkg/cmd/roachtest/tests/mixed_version_backup.go

```diff
@@ -2441,7 +2445,7 @@ func (mvb *mixedVersionBackup) verifyAllBackups(
 			}
 
 			if _, ok := collection.btype.(*clusterBackup); ok {
-				err := u.resetCluster(ctx, l, version, h.ExpectDeaths)
+				err := u.resetCluster(ctx, l, version, h.ExpectDeaths, mixedversion.ClusterSettingsForUpgrade(h.Context().FromVersion, h.Context().ToVersion))
```

Comment:
Hrmm, I wonder if exposing this function is worth it. We could either unconditionally pass the env var or compute whether we need the env var locally using `h.IsSkipVersionUpgrade`.

Reason I say this is because it reduces (even if a little) the API we expose. And backup-restore is pretty unique and the only test that does this kind of restart.

If you do want to keep this logic, we should fix it to just pass the env var when actually needed. Currently, it _seems_ that the code is doing that, but it's also passing the env var when the `version` argument to the `verify` function is `ToVersion`, and that's unnecessary.

Reply:
SGTM! Reverted exposing the function and went with `h.IsSkipVersionUpgrade` approach.